proposals/p1964.md

```diff
@@ -0,0 +1,282 @@
+
+### Types
+
+Since we will be representing the character literal as a sequence consisting of
+elements that are each either a unicode code-point or a hex-encoded byte value.
+
+We can see whether the particular literal is represented in the variable's type
+by only looking at the types:
+
+```
+let allowed: u8 = 'a';
+```
+
+The above is allowed because the type of 'a' indicates that it can be converted
+to `u8`.
+
+```
+let error1: u8 = '😃';
+let error2: u8 = 'AB';
+```
+
+However these should produce errors, the types of '😃' and 'AB' indicate that
+they cannot be converted to the declared type `u8`. However any `'\n'` and
+`'\u{A}'` would be of the same type, as they are the same unicode entities
+`%0A`. It is worth noting that `'\x0A'` will be slightly different as any
+character following the `\x...` sequence representation are not the same as a
+Unicode character. For example:
+
+```
+var c1: EBCDICChar = '\x0A';
+var c2: EBCDICChar = '\n';
+```
+
+Here we assume that `c1` will be the EBCDIC RPT character (0xA) and for `c2` to
+be the EBCDIC NL character (0x15)
+
+### Operations
+
+From the previous examples, we should be able to use the following operators:
+
+-   Comparison: `<`, `>`, `<=`, `>=` `==`
+    -   If any of the comparison operators are used between a code point and a
+        code unit, this will produce and error.
+-   Plus: `+`. However the behaviour of this operator is distinctly compared to
+    its use with Strings. Rather then a concatenation, this will add the value
+    of the two characters:
+    -   If the `+` is used between a character literal and an integer literal,
+        this should produce a character literal if the result fits into a
+        Unicode code point or for a unicode hex code like `'\xAB'`, a single
+        byte.
+    -   If the `+` is used between a character literal and an integer
+        non-literal this will produce and error.
+    -   If the `+` is used between two character literals this will produce and
+        error.
+    -   If the `+` operator is used between a code point and a code unit, this
+        will produce and error.
+-   Subtract: `-` Similar to the plus operator, this will subtract the value of
+    the two characters.
+
+    -   If the `-` is used between a character literal and an integer literal,
+        this should produce a character literal.
+    -   If the `-` is used between two character literals this should produce
+        and integer literal.
+    -   If the `-` is used between a character literal and an integer
+        non-literal this will produce and error.
+    -   If the `-` operator is used between a code point and a code unit, this
+        will produce and error.
+
+    In other use cases where the `+` and `-` operators are used, in theory we
+    should convert the character literal to some other type first. For example
+    if we want to call `x - 'a'`, where `w` is of type `char` we would want to
+    convert the `'a'` literal to a `char`.
```

Comment:
```suggestion
-   Comparison: `<`, `>`, `<=`, `>=` `==`
    -   If any of the comparison operators are used between a code point and a
        code unit, this will produce an error.
-   Plus: `+`. However the behaviour of this operator is distinct compared to
    its use with Strings. Rather than a concatenation, this will add the value
    of the two characters:
    -   If the `+` is used between a character literal and an integer literal,
        this should produce a character literal if the result fits into a
        Unicode code point or for a unicode hex code like `'\xAB'`, a single
        byte.
    -   If the `+` is used between a character literal and an integer
        non-literal this will produce an error.
    -   If the `+` is used between two character literals this will produce an
        error.
    -   If the `+` operator is used between a code point and a code unit, this
        will produce an error.
-   Subtract: `-` Similar to the plus operator, this will subtract the value of
    the two characters.

    -   If the `-` is used between a character literal and an integer literal,
        this should produce a character literal.
    -   If the `-` is used between two character literals this should produce
        an integer literal.
    -   If the `-` is used between a character literal and an integer
        non-literal this will produce an error.
    -   If the `-` operator is used between a code point and a code unit, this
        will produce an error.

    In other use cases where the `+` and `-` operators are used, in theory we
    should convert the character literal to some other type first. For example
    if we want to call `x - 'a'`, where `w` is of type `char` we would want to
    convert the `'a'` literal to a `char`.
```

Just typographical things: `and` -> `an`, `then` -> `than`.